lib/advert/ui/pages/detail_page/detail.dart

```diff
@@ -18,12 +19,26 @@ class AdvertDetailPage extends HookConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final advert = ref.watch(advertProvider);
+    final advertId = QR.params['advertId'].toString();
+    final advertList = ref.watch(advertListProvider);
+    final advertPostersNotifier = ref.read(advertPostersProvider.notifier);
+    final logoNotifier = ref.read(advertPosterProvider.notifier);
+
+    final advert = advertList.whenOrNull(
+      data: (adverts) => adverts.firstWhereOrNull((a) => a.id == advertId),
+    );
```

Comment:
You might want to check the select from provider : https://riverpod.dev/fr/docs/concepts/reading#usage-de-select-pour-filtrer-les-rebuilds